src/player.cpp

```diff
@@ -11642,6 +11643,48 @@ std::vector<const item *> player::all_items_with_flag( const std::string &flag )
     } );
 }
 
+item player::item_with_best_of_quality( quality_id qid )
+{
+    int maxq = max_quality( qid );
+    auto items_with_quality = items_with( [qid]( const item & it ) {
+        return it.has_quality( qid );
+    } );
+    for( item *it : items_with_quality ) {
+        if( it->get_quality( qid ) == maxq ) {
+            return *it;
+            break;
+        }
+    }
+    return null_item_reference();
+}
+
+bool player::crush_frozen_liquid( item_location loc )
+{
+
+    player &u = g->u;
```

Comment:
If you use `g->u` anyway, why is this a non-static player function? What does `somebody.crush_frozen_liquid( ... )` have to do with the `somebody`?

I suggest you use the `this` object instead of the global `g->u` here. Or alternatively make this a free function that is not connected to the `player` class.

Reply:
The reasoning for the connection to the player is for the check for `u.has_quality( quality_id( "HAMMER" ) )` and `u.item_with_best_of_quality( quality_id( "HAMMER" ) )`. That requires a connection to the player class unless we want to send the player as a parameter to the function which would work but be (if anything) more spaghetti-y. I see your point about the global part though and will change that to use `this` rather than referring to that variable. Thanks!